microservice/index.js

```diff
@@ -75,16 +76,33 @@ app.post('/tree', async (req, res) => {
     if(duplicate !== null){
       res.status(200).json({ duplicate });
     } else {
-      const tree = await data.createTree( user.id, req.body.device_identifier, req.body);
-      console.log("created tree " + tree.uuid);
-      res.status(201).json({ tree });
+      if(config.useFieldDataService) {
+        // translate to field-data capture payload
+        const tree = req.body
+        const capture = { 
+          ...tree,
+          id: tree.uuid,
+          planter_id: tree.user_id
+        };
+        var options = {
+          method: 'POST',
+          uri: config.fieldDataURI + "captures",
+          body: capture,
+          json: true // Automatically stringifies the body to JSON
+        };
+        const fieldCapture = await rp(options);
+        console.log("created field data tree capture " + fieldCapture.id);
+        res.status(201).json({ fieldCapture });
+      } else {
+        const tree = await data.createTree( user.id, req.body.device_identifier, req.body);
+        console.log("created tree " + tree.uuid);
+        res.status(201).json({ tree });
+      }
     }
 });
 
 app.put('/device', async (req, res) => {
-
     const device = await data.upsertDevice(req.body);
-    console.log("upsert device " + device.id);
```

Comment:
the call to `upsertDevice` doesn't really send back an object with `id` attribute.